Code/network/include/Resolver.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include "Network.h"
+#include "Endpoint.h"
+
+#include <string>
+#include <vector>
+#include <future>
+
+class Resolver
+{
+public:
+    class Iterator {
+    public:
+        using difference_type = ptrdiff_t;
+        using value_type = Endpoint;
+        using pointer = const Endpoint*;
+        using reference = const Endpoint&;
+        using iterator_category = std::forward_iterator_tag;
+
+        Iterator() : m_iterator() {}
```

Comment:
Should these braces go on new lines too?

Reply:
No it's fine for empty stuff like that